docs/new-architecture-app-renderer-ios.md

```diff
@@ -28,20 +28,32 @@ target 'Some App' do
 end
 
 def pods()
-  # Get config
-  config = use_native_modules!
-
-  # Use env variables to turn it on/off.
-  fabric_enabled = ENV['USE_FABRIC']
-
-  use_react_native!(
-    ...
-    # Modify here if your app root path isn't the same as this one.
-    :app_path => "#{Dir.pwd}/..",
-    # Pass the flag to enable fabric to use_react_native!.
-    :fabric_enabled => fabric_enabled
-  )
+# Get config
+config = use_native_modules!
+
+# Use env variables to turn it on/off.
+fabric_enabled = ENV['USE_FABRIC']
+use_codegen_discovery= ENV['USE_CODEGEN_DISCOVERY']
+
+# Enabled codegen discovery. This will run the codegen at preinstall time.
+# Files are generated at {pod installation root}/build/generated/ios/
+if use_codegen_discovery
+  Pod::UI.puts "[Codegen] Building target with codegen library discovery enabled."
+  pre_install do |installer|
+    use_react_native_codegen_discovery!({
+        react_native_path: config[:reactNativePath],
+        # Modify here if your app root path isn't the same as this one.
+        app_path: "#{Dir.pwd}/..",
+        fabric_enabled: fabric_enabled,
+    })
+  end
 end
+
+# Pass the flag to enable fabric to use_react_native!.
+use_react_native!(
+  ...
+  :fabric_enabled => fabric_enabled
+)
```

Comment:
Please rebase (or amend) as those changes I believe are not relevant.

Reply:
Done!